apache2/re.c

```diff
@@ -326,14 +326,14 @@ char *update_rule_target_ex(modsec_rec *msr, msre_ruleset *ruleset, msre_rule *r
                     if(value != NULL && targets[i]->param != NULL)  {
                         if((strlen(targets[i]->param) == strlen(value)) &&
                                 strncasecmp(targets[i]->param,value,strlen(targets[i]->param)) == 0) {
-                            memset(targets[i]->name,0,strlen(targets[i]->name));
-                            memset(targets[i]->param,0,strlen(targets[i]->param));
+                            targets[i]->name[0] = '\0';
```

Comment:
With this evaluation the string will be cleared - was this the intention? Based on old version it seems like the string's length is more than 0.

Reply:
memset sets all characters to '\0', which is useless, only the first character needs to be set to have a 0-ended string.